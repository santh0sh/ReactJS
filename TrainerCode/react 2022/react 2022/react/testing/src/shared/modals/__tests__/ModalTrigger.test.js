import { shallow, mount } from 'enzyme';
import ModalTrigger from '../ModalTrigger';

describe('Render ModalTrigger', () => {
    it('render component correctly', () => {
        const ModalTriggerComponent = shallow(<ModalTrigger><div /></ModalTrigger>);
        expect(ModalTriggerComponent).toMatchSnapshot();
    });

    describe('props and state behaviour', () => {
        const ModalTriggerComponent = mount(<ModalTrigger><div /></ModalTrigger>);

        it('ensure to have only one child (control element)', () => {
            expect(ModalTriggerComponent.findWhere(node => node.key() === 'modal-control').length).toEqual(1);
        });

        it('check children prop type', () => {
            expect(ModalTriggerComponent.props().children).toBeObject();
        });

        it('check the modal is opened', () => {
            const event = {
                preventDefault: () => {},
                stopPropagation: () => {}
            };
            ModalTriggerComponent.instance().open(event);
            expect(ModalTriggerComponent.state().toggled).toBeTruthy();
        });

        it('check the modal is closed', () => {
            ModalTriggerComponent.instance().close();
            expect(ModalTriggerComponent.state().toggled).toBeFalsy();
        });
    });
});
