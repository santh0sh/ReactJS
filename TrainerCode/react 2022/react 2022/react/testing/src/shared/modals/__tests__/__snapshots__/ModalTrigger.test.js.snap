// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render ModalTrigger render component correctly 1`] = `
Array [
  <div
    key="modal-control"
    onClick={[Function]}
  />,
  <ModalWrapper
    key="modal-dialog"
    onHide={[Function]}
    show={false}
  />,
]
`;
